src/tests/unit_x509.cpp

```diff
@@ -1592,8 +1593,885 @@ Test::Result test_x509_tn_auth_list_extension_decode() {
+
+      const auto& asnum = identifier.asnum().value().ranges().value();
+      result.confirm("cert has no RDI entries", identifier.rdi().has_value(), false);
+
+      // contains 0-999, 0-4294967295
+      result.confirm("asnum entry 0 min", asnum[0].min() == 0, true);
+      result.confirm("asnum entry 0 max", asnum[0].max() == 4294967295, true);
+   }
+   {
+      const std::string filename("ASRdiOnly.pem");
+      Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
+
+      auto as_blocks = cert.v3_extensions().get_extension_object_as<ASBlocks>();
+
+      const auto& identifier = as_blocks->as_identifiers();
+      result.confirm("cert has ASBlock extension", as_blocks != nullptr, true);
+
+      result.confirm("cert has no ASNUM entries", identifier.asnum().has_value(), false);
+      const auto& rdi = identifier.rdi().value().ranges().value();
+
+      // contains 1234-5678, 0-4294967295
+      result.confirm("rdi entry 0 min", rdi[0].min() == 0, true);
+      result.confirm("rdi entry 0 max", rdi[0].max() == 4294967295, true);
+   }
+   {
+      const std::string filename("ASNumberInherit.pem");
+      Botan::X509_Certificate cert(Test::data_file("x509/x509test/" + filename));
+
+      auto as_blocks = cert.v3_extensions().get_extension_object_as<ASBlocks>();
+
+      const auto& identifier = as_blocks->as_identifiers();
+      result.confirm("cert has ASBlock extension", as_blocks != nullptr, true);
+
+      result.confirm("asnum has no entries", identifier.asnum().value().ranges().has_value(), false);
+      const auto& rdi = identifier.rdi().value().ranges().value();
+
+      // contains 1234-5678, 0-4294967295
+      result.confirm("rdi entry 0 min", rdi[0].min() == 0, true);
+      result.confirm("rdi entry 0 max", rdi[0].max() == 4294967295, true);
+   }
+
+   result.end_timer();
+   return result;
+}
+
    #endif
 
+Test::Result test_x509_ip_addr_blocks_extension_encode() {
+   Test::Result result("X509 IP Address Block encode");
+   #if defined(BOTAN_HAS_RSA)
+   result.start_timer();
+
+   using Botan::Cert_Extension::IPAddressBlocks;
+
+   auto rng = Test::new_rng(__func__);
+
+   const std::string sig_algo{"RSA"};
+   const std::string hash_fn{"SHA-256"};
+   const std::string padding_method{"EMSA3(SHA-256)"};
+
+   for(size_t i = 0; i < 64; i++) {
+      bool push_ipv4_ranges = i & 1;
+      bool push_ipv6_ranges = i >> 1 & 1;
+      bool inherit_ipv4 = i >> 2 & 1;
+      bool inherit_ipv6 = i >> 3 & 1;
+      bool push_ipv4_family = i >> 4 & 1;
+      bool push_ipv6_family = i >> 5 & 1;
+
+      auto ca_key = make_a_private_key(sig_algo, *rng);
+      result.require("CA key", ca_key != nullptr);
+      const auto ca_cert = Botan::X509::create_self_signed_cert(ca_opts(), *ca_key, hash_fn, *rng);
+      Botan::X509_CA ca(ca_cert, *ca_key, hash_fn, padding_method, *rng);
+
+      auto key = make_a_private_key(sig_algo, *rng);
```

Comment:
There doesn't seem any need to create a new CA key, CA cert, or subject key for each iteration of the loop